pkg/credentialprovider/aws/aws_credentials_test.go

```diff
@@ -35,14 +37,27 @@ const email = "not@val.id"
 
 // Mock implementation
 type testTokenGetter struct {
-	user     string
-	password string
-	endpoint string
+	user              string
+	password          string
+	endpoint          string
+	randomizePassword bool
 }
 
-func (p *testTokenGetter) GetAuthorizationToken(input *ecr.GetAuthorizationTokenInput) (*ecr.GetAuthorizationTokenOutput, error) {
+type testTokenGetterFactory struct {
+	getter tokenGetter
+}
+
+func (f *testTokenGetterFactory) GetTokenGetterForRegion(region string) (tokenGetter, error) {
+	return f.getter, nil
+}
 
+func (p *testTokenGetter) GetAuthorizationToken(input *ecr.GetAuthorizationTokenInput) (*ecr.GetAuthorizationTokenOutput, error) {
+	if p.randomizePassword {
```

Comment:
Not sure if this really increases the test coverage that much (vs a fixed password like `secret`)

Reply:
This was added to check that you are actually pulling from the cache the second time since the password should be the same.

Reply:
Ah - clever.  A comment to that effect would make it obvious for the next reader!